source/common/request_info/dynamic_metadata_impl.cc

```diff
@@ -0,0 +1,53 @@
+#include "common/request_info/dynamic_metadata_impl.h"
+
+#include "envoy/common/exception.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+DynamicMetadataImpl::~DynamicMetadataImpl() {
+  for (auto& it : data_storage_) {
+    if (it.second.destructor_) {
+      it.second.destructor_(it.second.ptr_);
+    }
+  }
+  data_storage_.clear();
```

Comment:
Won't the implicit destructor do this for you?

Reply:
Yeah, this is just my discomfort with leaving pointers into freed memory around for even a little bit of time after they're freed.  But agreed, there's really no point.  Removed.